pynamodb/attributes.py

```diff
@@ -478,6 +482,44 @@ def _instantiate(cls: Type[_ACT], attribute_values: Dict[str, Dict[str, Any]]) -
         AttributeContainer._container_deserialize(instance, attribute_values)
         return instance
 
+    def to_dynamodb_dict(self) -> Dict[str, Any]:
+        """
+        Returns the contents of this instance as a JSON-serializable mapping,
+        using the same format as the "DynamoDB" JSON mapping in the AWS Console.
+        """
+        attr_values = self._container_serialize(null_check=False)
+        for v in attr_values.values():
+            bin_encode_attr(v)
+        return attr_values
+
+    def from_dynamodb_dict(self, d: Dict[str, Any]) -> None:
+        """
+        Sets attributes from a mapping previously produced by :func:`to_dynamodb_dict`.
+        """
+        for v in d.values():
+            bin_decode_attr(v)
+        self._update_attribute_types(d)
+        self._container_deserialize(d)
+
+    def to_normal_dict(self, *, force: bool = False) -> Dict[str, Any]:
+        """
+        Returns the contents of this instance as a normal JSON-serializable mapping,
+        using the same format as the "normal" JSON mapping in the AWS Console.
+
+        :param force: If :code:`True`, force the conversion even if the model contains Binary or Set attributes
+          (as they cannot be unambiguously represented in JSON, so this assumes the same model will be used
+          when parsing). If :code:`False`, a :code:`ValueError` will be raised if such attributes are set.
+        """
+        return {k: attr_value_to_normal_dict(v, force) for k, v in self._container_serialize(null_check=False).items()}
+
+    def from_normal_dict(self, d: Dict[str, Any]) -> None:
```

Comment:
to nit a little bit on the name, I wonder if the term "normal" might be a little ambiguous.. but I also don't necessarily have a better suggestion that is not horribly verbose. "simple" is maybe the only term that comes to mind, since it implies that there may be something unsupported

Reply:
I used "simple" before. Can switch back.

Reply:
0dc40c1795028763617785e455ecdb9a3669d8c2